### PR TITLE
76765-added Correlation id in logs for app insights

### PR DESCRIPTION
**Pull request by Nadeem14456**
In logging, there is no need to explicitly specify the correlation id.
The correlation id will be stored in the @CorrelationId field in AppInsights Trace Information under custom properties.
image

### CODE REVIEW

File: src/UKHO.ERPFacade.API/Filters/CorrelationIdMiddleware.cs [hunk 1]
```diff
@@ -1,28 +1,53 @@
...
+
+        public async Task InvokeAsync(HttpContext httpContext)
+        {
+            string? correlationId = string.Empty;
+
+            if (httpContext.Request.ContentLength > 0)
+            {
+                httpContext.Request.EnableBuffering();
+
+                var bodyAsText = await new StreamReader(httpContext.Request.Body).ReadToEndAsync();
```
Comment (jrippington) [inline src/UKHO.ERPFacade.API/Filters/CorrelationIdMiddleware.cs:25]: StreamReader is disposable. This line should be something like:

                using var streamReader = new StreamReader(httpContext.Request.Body, leaveOpen: true);
                var bodyAsText = await streamReader.ReadToEndAsync();

File: src/UKHO.ERPFacade.API/Filters/CorrelationIdMiddleware.cs [hunk 2]
```diff
@@ -1,28 +1,53 @@
...
+                JObject bodyAsJson = JObject.Parse(bodyAsText);
+                correlationId = bodyAsJson.SelectToken("data.traceId")?.Value<string>();
```
Comment (jrippington) [inline src/UKHO.ERPFacade.API/Filters/CorrelationIdMiddleware.cs:27]: Token id should be defined as a constant

File: src/UKHO.ERPFacade.API/Filters/LogCorrelationIdMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                //["CorrelationId"] = correlationId,
+            };
+
+            using (logger.BeginScope(state))
+            {
+                await _next(httpContext);
+            }
+        }
+
+
```
Comment (harshal11869) [inline src/UKHO.ERPFacade.API/Filters/LogCorrelationIdMiddleware.cs:55]: Please remove this blanks.

File: src/UKHO.ERPFacade.API/Filters/LogCorrelationIdMiddleware.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        public async Task InvokeAsync(HttpContext httpContext)
+        {
+            string? correlationId = string.Empty;
+
+            if (httpContext.Request.ContentLength > 0)
+            {
+                httpContext.Request.EnableBuffering();
+                var bodyAsText = await new StreamReader(httpContext.Request.Body).ReadToEndAsync();
+                JObject json = JObject.Parse(bodyAsText);
+                correlationId = json.SelectToken("data.traceId")?.Value<string>();
```
Comment (jrippington) [inline src/UKHO.ERPFacade.API/Filters/LogCorrelationIdMiddleware.cs:29]: Token id should be a constant

**Review by jrippington — CHANGES_REQUESTED** [review]
Few things to look at, looks good otherwise, thanks!

File: src/UKHO.ERPFacade.API/Filters/CorrelationIdMiddleware.cs (hunk 1 above)
Comment (Nadeem14456) [reply src/UKHO.ERPFacade.API/Filters/CorrelationIdMiddleware.cs]: The code has been refactored.

File: src/UKHO.ERPFacade.API/Filters/CorrelationIdMiddleware.cs (hunk 2 above)
Comment (Nadeem14456) [reply src/UKHO.ERPFacade.API/Filters/CorrelationIdMiddleware.cs]: I have made the necessary changes.

**Comment by Nadeem14456** [comment]
> Do we need @ as prefix with CorrelationId as highlighted in description ?

It is not required. I had previously created @CorrelationId as a custom field, but it has now been changed to _X-Correlation-ID.

**Review by harshal11869 — APPROVED** [review]
Looks Good.

File: src/UKHO.ERPFacade.API/Filters/LogCorrelationIdMiddleware.cs (hunk 4 above)
Comment (Nadeem14456) [reply src/UKHO.ERPFacade.API/Filters/LogCorrelationIdMiddleware.cs:29]: LogCorrelationIdMiddleware.cs is no longer in use. functionality from this is merged into CorrelationIdMiddleware.cs.

File: src/UKHO.ERPFacade.API/Filters/LogCorrelationIdMiddleware.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        }
+
+        public async Task InvokeAsync(HttpContext httpContext)
+        {
+            string? correlationId = string.Empty;
+
+            if (httpContext.Request.ContentLength > 0)
+            {
+                httpContext.Request.EnableBuffering();
+                var bodyAsText = await new StreamReader(httpContext.Request.Body).ReadToEndAsync();
```
[the remark this answers is not in the document]
Comment (Nadeem14456) [reply src/UKHO.ERPFacade.API/Filters/LogCorrelationIdMiddleware.cs:27]: LogCorrelationIdMiddleware.cs is no longer in use. functionality from this is merged into CorrelationIdMiddleware.cs.

**Review by jrippington — APPROVED** [review]
Looks good, thanks!

File: src/UKHO.ERPFacade.API/Filters/CorrelationIdMiddleware.cs
[the remark this answers is not in the document]
Comment (Nadeem14456) [reply src/UKHO.ERPFacade.API/Filters/CorrelationIdMiddleware.cs:1]: added unit Tests

File: src/UKHO.ERPFacade.API/Filters/LogCorrelationIdMiddleware.cs
[the remark this answers is not in the document]
Comment (Nadeem14456) [reply src/UKHO.ERPFacade.API/Filters/LogCorrelationIdMiddleware.cs:1]: file is deleted.

**Review by Shirin14926 — APPROVED** [review]
LGTM. Thanks @Nadeem14456

**Review by JiviteshT — APPROVED** [review]
Approved to progress merge process.

**Pull request merged**